pkg/exporters/stackdriver/stackdriver_exporter.go

```diff
@@ -0,0 +1,177 @@
+/*
+Copyright 2019 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package stackdriverexporter
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"reflect"
+	"time"
+
+	"contrib.go.opencensus.io/exporter/stackdriver"
+	monitoredres "contrib.go.opencensus.io/exporter/stackdriver/monitoredresource"
+	"github.com/golang/glog"
+	"github.com/spf13/pflag"
+	"go.opencensus.io/stats/view"
+	"google.golang.org/api/option"
+
+	"github.com/avast/retry-go"
+	"k8s.io/node-problem-detector/pkg/exporters"
+	seconfig "k8s.io/node-problem-detector/pkg/exporters/stackdriver/config"
+	"k8s.io/node-problem-detector/pkg/types"
+	"k8s.io/node-problem-detector/pkg/util/metrics"
+)
+
+func init() {
+	clo := commandLineOptions{}
+	exporters.Register(exporterName, types.ExporterHandler{
+		CreateExporterOrDie: NewExporterOrDie,
+		Options:             &clo})
+}
+
+const exporterName = "stackdriver"
+
+var NPDMetricToSDMetric = map[metrics.MetricID]string{
+	metrics.HostUptimeID:      "compute.googleapis.com/guest/system/uptime",
+	metrics.ProblemCounterID:  "compute.googleapis.com/guest/system/problem_count",
+	metrics.DiskAvgQueueLenID: "compute.googleapis.com/guest/disk/queue_length",
+	metrics.DiskIOTimeID:      "compute.googleapis.com/guest/disk/io_time",
+	metrics.DiskWeightedIOID:  "compute.googleapis.com/guest/disk/weighted_io_time",
+}
+
+func getMetricType(view *view.View) string {
+	viewName := view.Measure.Name()
+	// When there is no pre-defined Stackdriver Metric Type, fallback to custom metrics.
+	// e.g. custom.googleapis.com/npd/host/uptime
+	fallbackMetricType := "custom.googleapis.com/npd/" + viewName
```

Comment:
If custom metrics can incur extra cost, shall we make this configurable and default to drop them instead of using custom metrics?

Reply:
SG. Done.